### ISSUE TITLE
Favicon stays in light theme despite color scheme set to dark when privacy.resistFingerprinting is enabled

### ISSUE THREAD

**Issue by k4yt3x** [mbnuqw/sidebery#1790]
### Steps to reproduce

- Open Firefox in a new profile
- Install Sidebery
- Set Firefox theme to Dark in `about:addons` 
- Set Sidebery `Settings > Appearance > Color scheme`  to `dark`
- Enable `privacy.resistFingerprinting` in `about:config`
- Restart Firefox
- Open GitHub and observe its favicon color to be dark (light themed)

![image](https://github.com/user-attachments/assets/95abb4e1-c4b1-4c0a-b9c6-f76dfec72246)

### Actual behavior

When both Firefox and Sidebery's color themes are set explicitly set to dark, the auto-detection of color scheme mode for favicons in the sidebar still seems to read color scheme information using `prefers-color-scheme` or something similar, which gets blocked by `privacy.resistFingerprinting`. This causes certain websites' icons to be unreadable against the dark background.

### Expected behavior

It is understandable that when `Color scheme` is set to `auto light/dark` or `firefox` that it reads the wrong information from `privacy.resistFingerprinting` (I've read https://github.com/mbnuqw/sidebery/issues/997). However, when `Color scheme` is explicitly set to `dark` or `light`, the icon color theme should follow the explicitly defined Sidebery settings to avoid showing the wrong color theme.

### System

Arch Linux and Windows 11 23H2

### Firefox version

130.0 (archlinux - 1.0 and Windows)

### Sidebery version

5.2.0.8

### Logs

```code
Doesn't seem necessary at the moment. I can provide if it turns out to be needed.
```

**Comment by k4yt3x** [comment]
Thanks for the great extension btw! Can't use Firefox without it. It'll be great if this can be addressed so it's possible to use this with resist fingerprinting enabled.

**Comment by nnlq** [comment]
I second that

**Comment by k4yt3x** [comment]
Circled back to this in an attempt to figure out what's going on and maybe submitting a PR if it's fixable. Below is what I found:

GitHub's site favicon has `js-site-favicon` which indicates it might be set by a JS:

<img width="822" height="271" alt="Image" src="https://github.com/user-attachments/assets/41e0568f-81ae-42eb-809d-485d1b148ebc" />

Looking through the scripts this behavior one does reference `prefers-color-scheme`:

<img width="1483" height="697" alt="Image" src="https://github.com/user-attachments/assets/59173a3f-b262-4950-9c3a-51716a1d3bd0" />

Here's the code formatted:

<img width="1525" height="1239" alt="Image" src="https://github.com/user-attachments/assets/048c3959-7fe1-4360-8deb-72d85eaeafa9" />

**Basically GitHub uses JavaScript to determine if dark theme is preferred and sets the favicon to the dark variant if `prefers-color-scheme` is `dark`.** There are two variants of the favicon:

- https://github.githubassets.com/favicons/favicon.svg
- https://github.githubassets.com/favicons/favicon-dark.svg

It's not a Sidebery issue. Nor is this something Sidebery can/should address. I'm therefore closing this issue.

**Comment by k4yt3x** [comment]
For anyone else coming to this thread, if you really wanna keep both RFP and still get the dark icon here's one not-that-elegant-but-better-than-unreadable-way. Just use a script (that I AI'd together cuz I'm not frontend dev):

```javascript
// ==UserScript==
// @name         GitHub Favicon Dark Mode Fixer
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  Force dark favicon on GitHub when using dark Sidebery
// @match        https://github.com/*
// @match        https://gist.github.com/*
// @run-at       document-idle
// @grant        none
// ==/UserScript==

(function() {
    'use strict';

    function fixFavicon() {
        const faviconSvg = document.querySelector('.js-site-favicon[type="image/svg+xml"]');
        const faviconPng = document.querySelector('.js-site-favicon[type="image/png"]');

        if (faviconSvg && faviconPng) {
            const baseHref = faviconSvg.getAttribute('data-base-href');

            if (baseHref) {
                // Force dark variant
                faviconSvg.href = `${baseHref}-dark.svg`;
                faviconPng.href = `${baseHref}-dark.png`;
                console.log('Fixed GitHub favicon to dark variant');
            }
        }
    }

    // Run immediately
    fixFavicon();

    // Also watch for changes (GitHub uses Turbo for navigation)
    const observer = new MutationObserver(() => {
        fixFavicon();
    });

    observer.observe(document.head, {
        childList: true,
        subtree: true
    });

    // Listen for Turbo navigation events
    document.addEventListener('turbo:load', fixFavicon);
    document.addEventListener('turbo:render', fixFavicon);
})();
```
